.github/workflows/build-and-test.yml

```diff
@@ -1,21 +1,17 @@
 name: Build and test from latest commit on master
-
 on:
   push:
     branches: [ master ]
   pull_request:
     branches: [ master ]
-
 jobs:
-
   build_and_test:
     name: Build and test on ${{ matrix.os }}
     runs-on: ${{ matrix.os }}
     strategy:
```

Comment:
since you are not going to use different OSes, you can simplify this portion of the GHA workflow

Reply:
Made changes as mentioned.